pkg/awsutils/awsutils.go

```diff
@@ -870,7 +870,13 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddresses(eniID string, numIPs int
 	var needIPs = numIPs
 
 	ipLimit, err := cache.GetENIipLimit()
-	if err == nil && ipLimit < needIPs {
+	if err != nil {
+		awsUtilsErrInc("UnknownInstanceType", err)
+		// for unknown instance type, will allocate one ip address at a time
+		ipLimit = 1
```

Comment:
On error, we should probably just return it, same as for `getMaxEni()`. I added #485 to remove the unused code where the 1 came from. We shouldn't default to 1 IP, that won't work.

Reply:
Btw to clarify. We should call `awsUtilsErrInc("UnknownInstanceType", err)`, but after that just `return err`.

Reply:
@mogren Done